### ISSUE TITLE
@select returns multiple values without the store values actually changing

### ISSUE THREAD

**Issue by marcinkunert** [angular-redux2/store#3]
### This is a bug report

I've just switched from `@angular-redux/store": "~10.0.0"` to ` @angular-redux2/store": "~2.1.3",`.
I did no changes to the code except changing the imports.
Following code with @select started to return 2 values (first `false` then `true`). The previous store returned only `true`.

```
@Injectable()
export class IsLoggedInAuthGuard implements CanActivate {

  @select([ 'auth', 'isLoggedIn' ]) isLoggedIn$: Observable<boolean>;

  constructor(private router: Router,
              private authActions: AuthActions) {}

  canActivate(route: ActivatedRouteSnapshot,
              state: RouterStateSnapshot): Observable<boolean> {

    return this.isLoggedIn$
      .pipe(
        tap(data => console.log('IsLoggedInAuthGuard tap', data)),
        tap(logged => {
          if (!logged) {
            this.authActions.saveRedirectUrl(state.url);
            this.router.navigate([ '/logowanie' ]);

            console.log('IsLoggedInAuthGuard', logged);
          }
        })
      );
  }
}
```

**Comment by Garefild** [comment]
Hi, there is some change in the lib, soon I will upload a new version with docs. 
in the mine time, I try to mimic the problem but could not create the problem, it could help me if you can give me more information about your structure. 

`app.component.ts`
```typescript

import { Component, OnInit } from '@angular/core';
import { Observable } from "rxjs";
import { Dispatch, select } from "@angular-redux2/store";
import { IS_LOGIN } from "./store/actions/auth.action";

@Component({
    selector: 'app-root',
    templateUrl: './app.component.html',
    styleUrls: [ './app.component.scss' ]
})
export class AppComponent implements OnInit {
    title = 'workspace';

    @select([ 'auth', 'isLoggedIn' ]) isLoggedIn$: Observable<boolean>;

    ngOnInit() {
        this.isLoggedIn$?.subscribe(r => {
            console.log(r);
        });
    }

    @Dispatch
    testClick() {
        return {
            type: IS_LOGIN,
        };
    }
}

```
`app.component.html`
```html
<h1>{{ isLoggedIn$ | async }}</h1>

<button (click)="testClick()">change state</button>
```

`store/store.module.ts`
```typescript
/**
 * Imports
 */

import { CommonModule } from '@angular/common';
import { isDevMode, NgModule } from '@angular/core';

/**
 * Imports actions
 */

import { IAppState, INITIAL_STATE, rootReducer } from './state';
import { DevToolsExtension, NgRedux, NgReduxModule } from "@angular-redux2/store";


/**
 * Redux store module
 */


@NgModule({
    declarations: [],
    imports: [
        CommonModule,
        NgReduxModule
    ]
})
export class StoreModule {

    /**
     * Constructor
     */

    constructor(ngRedux: NgRedux<IAppState>, devTools: DevToolsExtension) {
        let enhancer: Array<any> = [];

        if (devTools.enhancer() && isDevMode())
            enhancer = [ devTools.enhancer() ];

        ngRedux.configureStore(<any> rootReducer, INITIAL_STATE, [], enhancer);
    }
}
```


`store/state.ts`
```typescript
/**
 * Imports
 */

import { combineReducers, Reducer } from 'redux';

/**
 * Actions
 */

import { AUTH_INITIAL_STATE, authReducer, Auth } from "./actions/auth.action";

/**
 * IAppState interface
 */

export interface IAppState {
    auth: Auth,
}

/**
 * Init IAppState
 */

export const INITIAL_STATE: IAppState = {
    auth: AUTH_INITIAL_STATE,
};

/**
 * Root reducer
 */

export const rootReducer: Reducer<IAppState> = combineReducers({
    auth: authReducer,
});
```

> this is a class reducer you can write it in the old way.

`store/actions/auth.action.ts`
```typescript
/**
 * Imports
 */

import { AnyAction } from "redux";
import { Action, createReducerFromClass } from "@angular-redux2/store";

/**
 * Update auth user data
 */

export const IS_LOGIN = 'IS_LOGIN';

/**
 * AuthState interface
 */

export interface Auth {
    isLoggedIn: boolean;
}

/**
 * Init Auth model
 */

export const AUTH_INITIAL_STATE: Auth = {
    isLoggedIn: false
};

/**
 * Class Reducer
 */

class Reducer {
    /**
     * Update login data
     */

    @Action(IS_LOGIN)
    isLogin(state: Auth, action: AnyAction) {
        return { isLoggedIn: !state.isLoggedIn };
    }
}

export const authReducer = createReducerFromClass(Reducer, AUTH_INITIAL_STATE);
```

> old way

```typescript
/**
 * Imports
 */

import { AnyAction } from "redux";

/**
 * Update auth user data
 */

export const IS_LOGIN = 'IS_LOGIN';

/**
 * AuthState interface
 */

export interface Auth {
    isLoggedIn: boolean;
}

/**
 * Init Auth model
 */

export const AUTH_INITIAL_STATE: Auth = {
    isLoggedIn: false
};

/**
 * AuthReducer
 */

export function authReducer(state: Auth = AUTH_INITIAL_STATE, action: AnyAction): Auth {
    switch (action.type) {
        case IS_LOGIN:
            return { isLoggedIn: !state.isLoggedIn };
    }

    return state;
}
```

**Comment by marcinkunert** [comment]
I'll try to investigate that further if I find some time. In this issue, I just wanted to point out that there is something wrong going on. I've thought this might be something obvious, but (as always 😅) it's not.

**Comment by Garefild** [comment]
Maybe it works properly, you just do "Dispatch" twice?
If you could bring up some of the source code I could research the topic it would be great.

**Comment by Garefild** [comment]
Hi, So I found the problem, in the next few days I'll push a fix to git and npmjs. 
tnx for the report.

**Comment by Garefild** [comment]
fixed at version 4.0.2.
